### ISSUE TITLE
 Error: Unknown named parameter $stats

### ISSUE THREAD

**Issue by MannikJ** [madnest/madzipper#22]
The following parts of the code throw an error with php 8 because the implementation passes the named parameter `$stats` which is neither defined in the methods signature nor used. 

> Error: Unknown named parameter $stats
> 
> xqz/vendor/madnest/madzipper/src/Madnest/Madzipper/Repositories/ZipRepository.php:119
> xyz/vendor/madnest/madzipper/src/Madnest/Madzipper/Madzipper.php:501

Madzipper:
```php
    public function listFiles($regexFilter = null)
    {
        $filesList = [];
        if ($regexFilter) {
            $filter = function ($file) use (&$filesList, $regexFilter) {
                // push/pop an error handler here to to make sure no error/exception thrown if $expected is not a regex
                set_error_handler(function () {
                });
                $match = preg_match($regexFilter, $file);
                restore_error_handler();

                if ($match === 1) {
                    $filesList[] = $file;
                } elseif ($match === false) {
                    throw new \RuntimeException("regular expression match on '$file' failed with error. Please check if pattern is valid regular expression.");
                }
            };
        } else {
            $filter = function ($file) use (&$filesList) {
                $filesList[] = $file;
            };
        }
        $this->repository->each($filter);

        return $filesList;
    }
 ```
 ZipRepositry:
 ```php
     public function each($callback)
    {
        for ($i = 0; $i < $this->archive->numFiles; ++$i) {
            //skip if folder
            $stats = $this->archive->statIndex($i);
            if ($stats['size'] === 0 && $stats['crc'] === 0) {
                continue;
            }
            call_user_func_array($callback, [
                'file' => $this->archive->getNameIndex($i),
                'stats' => $this->archive->statIndex($i),
            ]);
        }
    }
 ```

**Comment by MannikJ** [comment]
Just saw this is an duplicate of #14 which has already been fixed by a PR. But apparently it has not been released yet.

**Comment by theimerj** [comment]
Should be fixed with `v1.2.1` and `1.3.1` respectively. Thank you for pointing that out.
